### PR TITLE
Lab 9

### PR DESCRIPTION
**Pull request by YaroslavLys**

### CODE REVIEW

File: src/main/java/ua/lviv/iot/military_unit/App.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
+package ua.lviv.iot.military_unit;
+
+import ua.lviv.iot.military_unit.manager.ArmamentManager;
+import ua.lviv.iot.military_unit.manager.SortOrder;
+import ua.lviv.iot.military_unit.models.*;
```
Comment (Olezhko2001) [inline src/main/java/ua/lviv/iot/military_unit/App.java]: it's a bad practice to have * imports, please get rid of them

File: src/main/java/ua/lviv/iot/military_unit/manager/ArmamentManager.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            System.out.println("function: " + item.getFunction());
+            System.out.println("price: " + item.getPrice());
+            System.out.println("mortality rate: " + item.getMortalityRate());
+            System.out.println("---------");
+        }
+    }
+
+    public void sortByPrice(SortOrder order) {
+        if (order == SortOrder.ASC) {
+            this.arsenal.sort((Armament firstItem, Armament secondItem) ->
```
Comment (Olezhko2001) [inline src/main/java/ua/lviv/iot/military_unit/manager/ArmamentManager.java]: you don't have to mention data types in lambdas

File: src/main/java/ua/lviv/iot/military_unit/manager/ArmamentManager.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+                    firstItem.getMortalityRate() - secondItem.getMortalityRate());
+        } else {
+            this.arsenal.sort((Armament firstItem, Armament secondItem) ->
+                    secondItem.getMortalityRate() - firstItem.getMortalityRate());
+        }
+    }
+
+    public void searchByFunction(String function) {
```
Comment (Olezhko2001) [inline src/main/java/ua/lviv/iot/military_unit/manager/ArmamentManager.java:48]: why don't you return them, but change the original list?

File: src/main/java/ua/lviv/iot/military_unit/ArmamentController.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+            return Response.ok().build();
+        } else {
+            return Response.status(Response.Status.NOT_FOUND).build();
+        }
+    }
+
+    @POST
+    @Consumes(MediaType.APPLICATION_FORM_URLENCODED)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response createArmament(@FormParam("name") String name) {
```
Comment (Olezhko2001) [inline src/main/java/ua/lviv/iot/military_unit/ArmamentController.java:53]: what if you make a mistake in "name" field name? does it throw 4xx error code? if no, it should

Comment (YaroslavLys) [reply src/main/java/ua/lviv/iot/military_unit/ArmamentController.java:53]: The POST method is used to submit an entity to the specified resource, often causing a change in state or side effects on the server. In this code I use POST method only to create a new armament. I don't quite understand what do you mean.

File: src/main/java/ua/lviv/iot/military_unit/manager/ArmamentManager.java (hunk 2 above)
Comment (YaroslavLys) [reply src/main/java/ua/lviv/iot/military_unit/manager/ArmamentManager.java]: Will be fixed

File: src/main/java/ua/lviv/iot/military_unit/App.java (hunk 1 above)
Comment (YaroslavLys) [reply src/main/java/ua/lviv/iot/military_unit/App.java]: Will be fixed

File: src/main/java/ua/lviv/iot/military_unit/manager/ArmamentManager.java (hunk 3 above)
Comment (YaroslavLys) [reply src/main/java/ua/lviv/iot/military_unit/manager/ArmamentManager.java:48]: According to the task, further actions should be performed only with objects of certain function. This is one of the possible ways

**Pull request merged**